src/connectors/argentMobile/modal/starknet/adapter.ts

```diff
@@ -102,10 +107,60 @@ export class StarknetAdapter
   }
 
   // StarknetWindowObject
+  request: RequestFn = async (call): Promise<any> => {
+    if (!this.session) {
+      throw new Error("No session")
+    }
+
+    if (call.type === "wallet_requestChainId") {
+      return this.chainId === constants.NetworkName.SN_GOERLI
+        ? constants.StarknetChainId.SN_GOERLI
+        : constants.StarknetChainId.SN_MAIN
+    }
+
+    if (call.type === "wallet_requestAccounts") {
+      return this.accounts
+    }
+
+    if (call.type === "wallet_getPermissions") {
+      if (await this.isPreauthorized()) {
+        return ["accounts"]
+      }
+
+      return []
+    }
+
+    if (call.type === "starknet_addInvokeTransaction") {
+      const { calls } = call.params as AddInvokeTransactionParameters
+
+      return await this.requestWallet({
+        method: "starknet_requestAddInvokeTransaction",
+        params: {
+          accountAddress: this.account.address,
+          executionRequest: {
+            calls: calls?.map(({ contract_address, ...rest }) => ({
+              ...rest,
+              contractAddress: contract_address,
+            })),
+          },
+        },
+      })
+    }
```

Comment:
minor but maybe we can have a map here for a better readability?
something like:

const requestHandlers = Object.freeze({
  'starknet_addInvokeTransaction': handleAddInvokeTransaction,
  'wallet_getPermissions': handleWalletGetPermissions
  ....

instead of a bunch of if's